### PR TITLE
Add Carbon datasource for Hadoop

### PR DESCRIPTION
**Pull request by gokul**
This component will make it possible for the user to specify Carbon datasource definitions for connecting with Hadoop-like data sources (e.g. Hadoop, HBase) using the Hadoop Configuration format.

### CODE REVIEW

File: components/org.wso2.carbon.datasource.hadoop/src/main/java/org/wso2/carbon/datasource/hadoop/HadoopDataSourceUtil.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public static Configuration createConfigFromFile(String xmlConfiguration) throws DataSourceException {
+        try {
+            HadoopDataSourceConfiguration configuration = DataSourceUtils
+                    .loadJAXBConfiguration(xmlConfiguration, HadoopDataSourceConfiguration.class);
+
+            Configuration config = new Configuration();
+            List<HadoopDataSourceProperty> properties = configuration.getProperties();
+
+            for (HadoopDataSourceProperty configEntry : properties) {
+                if (!("".equals(configEntry.getPropertyName())) && !("".equals(configEntry.getPropertyValue()))) {
```
Comment (laki88) [inline components/org.wso2.carbon.datasource.hadoop/src/main/java/org/wso2/carbon/datasource/hadoop/HadoopDataSourceUtil.java]: We can use isEmpty()?

File: components/org.wso2.carbon.datasource.hadoop/src/test/resources/testng.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (laki88) [inline components/org.wso2.carbon.datasource.hadoop/src/test/resources/testng.xml]: License text should be in 2017

File: components/org.wso2.carbon.datasource.hadoop/pom.xml [hunk 3]
```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2016 WSO2, Inc. (http://wso2.com)
```
Comment (laki88) [inline components/org.wso2.carbon.datasource.hadoop/pom.xml]: License text should be 2017

File: components/org.wso2.carbon.datasource.hadoop/pom.xml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                            <nohelp>true</nohelp>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <properties>
+        <export.package>org.wso2.carbon.datasource.hadoop.*</export.package>
+        <import.package>javax.xml.parsers;version=0.0.0,*;resolution:=optional</import.package>
```
Comment (laki88) [inline components/org.wso2.carbon.datasource.hadoop/pom.xml]: Can't we get rid of this resolution optional?

**Comment by gokul** [comment]
Closing PR since this functionality is not needed just yet for DAS 4.

**Pull request closed without merging**